scanner/templates/laravel/.fly/nginx-default

```diff
@@ -0,0 +1,33 @@
+# HTTP configuration
+#
+server {
+    listen 8080 default_server;
+    listen [::]:8080 default_server;
+
+    root /var/www/html/public;
+
+    # Set allowed "index" files
+    index index.html index.htm index.php;
+
+    server_name _;
+
+    charset utf-8;
+
+    # Set max upload to 2048M
+    client_max_body_size 2048M;
+
+    # Have NGINX try searching for PHP files as well
+    location / {
+        try_files $uri $uri/ /index.php?$query_string;
+    }
+
+    # Pass "*.php" files to PHP-FPM
+    location ~ \.php$ {
+        include snippets/fastcgi-php.conf;
+        fastcgi_param HTTP_X_FORWARDED_FOR $http_fly_client_ip;
```

Comment:
I would suggest adding this line as well:

```
fastcgi_param HTTPS on;
```

Without this line, `$request->isSecure()` will return false. This affects how URLs are generated in Laravel/Symfony

Reply:
That's set already in `/etc/nginx/fastcgi.conf` which is itself included via `include snippets/fastcgi-php.conf`

Reply:
This line is specified as:

`fastcgi_param  HTTPS              $https if_not_empty;`

I am not too proficient in NGINX configurations but this passes the `$https` value as `HTTPS` header. From the docs:

```
$https
    “on” if connection operates in SSL mode, or an empty string otherwise
```

The connection to the VM does not operate in SSL mode so this will be an empty string I suppose?

I doubt this is working out of the box with `fastcgi-php.conf`. We were getting `$request->isSecure() === false` until we added `fastcgi_param HTTPS on;`

Reply:
Hm, I just created a new Laravel project, ran `fly launch` and `dd(request()->isSecure())` and it shows `true`. Nvm then, I guess there's something different between our configuration and the current config given by `fly launch`...

Reply:
Oddly enough this does not work in Symfony. I use the Laravel configuration for Symfony (with some adjustments), and `isSecure()` returns false if I don't add that `fastcgi_param HTTPS on;` line.

Reply:
I suspect Laravel is reading the `X-Forwarded-Proto` header to determine the result of `isSecure()` (since the TrustedProxy middleware is being used/enabled), and in that case, it likely ignores `HTTPS`.

That's a good point tho, Fly's proxy is sending `http://` requests to the VM after terminating the TLS connection 🤔